pkg/apis/core/validation/validation.go

```diff
@@ -2711,6 +2719,71 @@ func ValidateVolumeDevices(devices []core.VolumeDevice, volmounts map[string]str
 	return allErrs
 }
 
+func validatePodResourceClaims(claims []core.PodResourceClaim, fldPath *field.Path) (sets.String, field.ErrorList) {
+	var allErrs field.ErrorList
+	podClaimNames := sets.NewString()
+	for i, claim := range claims {
+		allErrs = append(allErrs, validatePodResourceClaim(claim, podClaimNames, fldPath.Index(i))...)
+	}
+	return podClaimNames, allErrs
+}
+
+func validatePodResourceClaim(claim core.PodResourceClaim, podClaimNames sets.String, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if claim.Name == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "a name is required to reference the claim"))
+	} else if podClaimNames.Has(claim.Name) {
+		allErrs = append(allErrs, field.Duplicate(fldPath.Child("name"), claim.Name))
+	} else {
+		podClaimNames.Insert(claim.Name)
+	}
+	allErrs = append(allErrs, validatePodResourceClaimSource(claim.Source, fldPath.Child("source"))...)
+
+	return allErrs
+}
+
+func validatePodResourceClaimSource(claimSource core.ClaimSource, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if claimSource.ResourceClaimName != nil && claimSource.TemplateName != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, claimSource, "either resourceClaimName or template must be set, not both"))
+	}
+	if claimSource.ResourceClaimName == nil && claimSource.TemplateName == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, claimSource, "either resourceClaimName or template must be set, both are unset"))
+	}
+	if claimSource.ResourceClaimName != nil {
+		for _, msg := range ValidateResourceClaimName(*claimSource.ResourceClaimName, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceClaimName"), *claimSource.ResourceClaimName, msg))
+		}
+	}
+	if claimSource.TemplateName != nil {
+		for _, msg := range ValidateResourceClaimTemplateName(*claimSource.TemplateName, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("templateName"), *claimSource.TemplateName, msg))
+		}
+	}
+	return allErrs
+}
+
+func validateResourceClaimNames(claims []core.ResourceClaim, podClaimNames sets.String, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	names := sets.NewString()
+	for i, claim := range claims {
+		name := claim.Name
+		if name == "" {
+			allErrs = append(allErrs, field.Required(fldPath.Index(i), "name cannot be empty"))
+		} else {
+			if names.Has(name) {
+				allErrs = append(allErrs, field.Duplicate(fldPath.Index(i), name))
+			} else {
+				names.Insert(name)
+			}
+			if !podClaimNames.Has(name) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(i), name, "not defined in pod.spec.resourceClaims"))
```

Comment:
VolumeMounts example is: `allErrs = append(allErrs, field.NotFound(idxPath.Child("name"), mnt.Name))`

Reply:
Changed.